odoo/addons/base/models/ir_actions_report.py

```diff
@@ -644,17 +644,24 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
 
         # access the report details with sudo() but evaluation context as current user
         report_sudo = self._get_report(report_ref)
+        has_duplicated_ids = res_ids and len(res_ids) != len(set(res_ids))
 
         collected_streams = OrderedDict()
 
         # Fetch the existing attachments from the database for later use.
         # Reload the stream from the attachment in case of 'attachment_use'.
         if res_ids:
             records = self.env[report_sudo.model].browse(res_ids)
-            for record in records:
+            record_by_id = {record.id: record for record in records}
+            for res_id in res_ids:
+                if res_id in collected_streams:
+                    collected_streams[res_id]['multiplicity'] += 1
+                    continue
```

Comment:
You don't need `record_by_id`. recordset is actually a recordlist.
```suggestion
            for record in records:
                if record.id in collected_streams:
                    collected_streams[record.id]['multiplicity'] += 1
                    continue
```

Reply:
@ryv-odoo I always forget the browse keeps the duplicated ids :woman_facepalming: